sync/src/types.rs

```diff
@@ -308,65 +386,26 @@ impl DownloadScheduler {
         self.task_count
     }
 
-    fn adjust(&mut self, time: u64, len: u64) {
-        let now = unix_time_as_millis();
-        // 8 means default max outbound
-        // All synchronization tests are based on the assumption of 8 nodes.
-        // If the number of nodes is increased, the number of requests and processing time will increase,
-        // and the corresponding adjustment criteria are needed, so the adjustment is based on 8.
-        //
-        // note: This is an interim scenario and the next step is to consider the median response time
-        // within a certain interval as the adjustment criterion
-        let (quotient, remainder) = if len > 8 { (len >> 3, len % 8) } else { (1, 0) };
-        // Dynamically adjust download tasks based on response time.
-        //
-        // Block max size about 700k, Under 10m/s bandwidth it may cost 1s to response
-        // But the mark is on the same time, so multiple tasks may be affected by one task,
-        // causing all responses to be in the range of reduced tasks.
-        // So the adjustment to reduce the number of tasks will be calculated by modulo 3 with the actual number of triggers.
-        // Adjust for each block received.
-        match ((now - time) / quotient).saturating_sub(remainder * 100) {
-            // Within 500ms of response, considered better to communicate with that node's network
-            0..=500 => {
-                if self.task_count < FIRST_LEVEL_MAX - 1 {
-                    self.task_count += 2
-                } else {
-                    self.breakthroughs_count += 1
-                }
-            }
-            // Within 1000ms of response time, the network is relatively good and communication should be maintained
-            501..=1000 => {
-                if self.task_count < FIRST_LEVEL_MAX {
-                    self.task_count += 1
-                } else {
-                    self.breakthroughs_count += 1
-                }
-            }
-            // Response time within 1500ms, acceptable range, but not relatively good for nodes and do not adjust
-            1001..=1500 => (),
-            // Above 1500ms, relatively poor network, reduced
-            _ => {
-                self.timeout_count += 1;
-                if self.timeout_count > 3 {
-                    self.task_count = self.task_count.saturating_sub(1);
-                    self.timeout_count = 0;
-                }
-            }
+    fn increase(&mut self, num: usize) {
+        if self.task_count < MAX_BLOCKS_IN_TRANSIT_PER_PEER {
+            self.task_count += num
         }
-        if self.breakthroughs_count > 4 {
-            if self.task_count < MAX_BLOCKS_IN_TRANSIT_PER_PEER {
-                self.task_count += 1;
-            }
-            self.breakthroughs_count = 0;
+    }
+
+    fn decrease(&mut self, num: usize) {
+        self.timeout_count += num;
+        if self.timeout_count > 2 {
```

Comment:
Why changed from 3 to 2? Why not decreasing `task_count` directly?

Reply:
Since it does not decrease `task_count` like what `increase` does, a name like `on_timeout` is better.

Reply:
> Why changed from 3 to 2? Why not decreasing `task_count` directly?

Experience value, it will affect efficiency after changing to direct operation

Reply:
> Since it does not decrease `task_count` like what `increase` does, a name like `on_timeout` is better.

I think the behavior is consistent

Reply:
By the way, the penalty and bonus values are matched when the penalty range hits about 1.5 times the bonus range's the `task_count` must be trending downward until it is finally expelled.